trio/_channel.py

```diff
@@ -1,18 +1,46 @@
 from collections import deque, OrderedDict
 from math import inf
 
+from types import TracebackType
+from typing import (
+    Generic,
+    TypeVar,
+    Any,
+    Tuple,
+    Optional,
+    Type,
+    Deque,
+    OrderedDict as T_OrderedDict,
+    Set,
+)
+
 import attr
 from outcome import Error, Value
 
 from .abc import SendChannel, ReceiveChannel, Channel
 from ._util import generic_function, NoPublicConstructor
 
 import trio
-from ._core import enable_ki_protection
+from ._core import enable_ki_protection, Task, Abort
+
+# A regular invariant generic type
+T = TypeVar("T")
+
+# The type of object produced by a ReceiveChannel (covariant because
+# ReceiveChannel[Derived] can be passed to someone expecting
+# ReceiveChannel[Base])
+ReceiveType = TypeVar("ReceiveType", covariant=True)
+
+# The type of object accepted by a SendChannel (contravariant because
+# SendChannel[Base] can be passed to someone expecting
+# SendChannel[Derived])
+SendType = TypeVar("SendType", contravariant=True)
```

Comment:
This feels a bit wrong but thinking about it, it looks right. (I made some code examples to prove it to myself, this variance stuff is really tricky).

Reply:
As I said in my first post I just assumed it'd be the same as the abc, so thanks for double checking it :)